packages/run-protocol/src/vpool-xyk-amm/addPool.js

```diff
@@ -29,32 +37,71 @@ export const makeAddIssuer = (
       E(secondaryIssuer).getAssetKind(),
       E(secondaryIssuer).getBrand(),
     ]);
+    // AWAIT ///////////////
 
-    assert(
-      !isInSecondaries(secondaryBrand),
-      X`issuer ${secondaryIssuer} already has a pool`,
-    );
     assert(
       secondaryAssetKind === AssetKind.NAT,
       X`${keyword} asset not fungible (must use NAT math)`,
     );
+
+    if (brandToLiquidityIssuer.has(secondaryBrand)) {
+      return brandToLiquidityIssuer.get(secondaryBrand);
+    }
+
+    assert(
+      !isInSecondaries(secondaryBrand),
+      X`issuer ${secondaryIssuer} already has a pool`,
+    );
+
+    if (brandToLiquidityMint.has(secondaryBrand)) {
+      const { issuer } = brandToLiquidityMint
+        .get(secondaryBrand)
+        .getIssuerRecord();
+      return issuer;
+    }
+
+    /** @template T @typedef {import('@endo/promise-kit').PromiseKit<T>} PromiseKit */
+    /** @type {PromiseKit<Issuer<'nat'>>} */
+    const promiseKit = makePromiseKit();
+
+    // If there's a race to add an issuer, one will succeed here and
+    // others will fail. The failures should get the promise.
+    try {
+      brandToLiquidityIssuer.init(secondaryBrand, promiseKit.promise);
+    } catch (e) {
```

Comment:
This is not the error case I was thinking of. I think this should not be possible because the `has` check is above in the same turn. The case I'm concerned about is if it fails below (e.g., in `saveIssuer` or  `addIssuerToReserve`).  Those would prevent the promise from being resolved on line 103.

Reply:
here's a fix.